### PR TITLE
Adds maint drone customisation

### PR DESCRIPTION
**Pull request by ExusA**
## What Does This PR Do
Adds in three new selectable drone skins, selectable when a player joins as a drone. Each of the four drone skins (including the original one) can be any combination of on, off, emagged, and on your head. The basic sprites are ports, but the hats and emagged sprites are my own, not amazing spritework. Support for drone fluff sprites is still in, (though admittedly as snowflakey as ever, and there were never any drone fluff sprites anyway) If para ever does fluff sprites again, it should still work the same way.

## Why It's Good For The Game
This has been requested a few times over a while, and generally just adds more flavor and fun for drone players.

## Images of changes
drones_front
From left to right: brass, prismatic, and pointy drone chassis, with the original for reference

drone_emag
The same, but emagged this time

drone_hats
The same, in hat form...

drone_emaghats
...And the cursed emag-hat combo.

(forgot to mention, moved all drone sprites from robots.dmi to the new drone.dmi to make things a bit cleaner)

## Changelog
:cl:
add: Added drone skins, with their own emagged and on-head sprites
/:cl:

### CODE REVIEW

**Comment by Spacemanspark** [comment]
Damn, these look dope!

**Comment by taukausanake** [comment]
The normal drone only has the eyes light up when emagged but these new drones have other features on them gain red when emagged. Looks a little weird. Maybe keep the emag icons to just the eyes? It doesn't look like they have other glowing features that would change

**Comment by ExusA** [comment]
the normal drones actually do have their yellow trim turn red when they're emagged, so you can see at a glance even when they're unconscious and the red eye-lights are off. Maybe I overdid it a bit, but when a drone is emagged, you should be able to tell pretty easily at a glance so you don't ahelp them and that sorta thing.

**Comment by ExusA** [comment]
well, i messed up git somehow, so have to redo a few things

**Comment by ExusA** [comment]
alrighty, seems to be working again. tentatively going to say this is ready for review...

File: code/modules/mob/living/silicon/robot/drone/drone.dm [hunk 1]
```diff
@@ -297,18 +292,16 @@
 	to_chat(src, "<b>Don't invade their worksites, don't steal their resources, don't tell them about the changeling in the toilets.</b>")
 	to_chat(src, "<b>Make sure crew members do not notice you.</b>.")
 
-/*
-	sprite["Default"] = "repairbot"
-	sprite["Mk2 Mousedrone"] = "mk2"
-	sprite["Mk3 Monkeydrone"] = "mk3"
+	//now that the drone has a player, let's pick a sprite.
+	var/drone_icons[0]	//these are our sprite choices
```
Comment (SteelSlayer) [inline code/modules/mob/living/silicon/robot/drone/drone.dm:296]: This should be `var/list/drone_icons = list()`

Or even better, you can do:

```js
var/list/drone_icons = list(
    "Standard" = "repairbot",
	"Pointy" = "drone_scout",
	"Prismatic" = "drone_gem",
	"Clockwork" = "drone_clock"
)

**Comment by Landerlow** [comment]
I love these! I love drones, this will make me love drones even more.

File: code/modules/mob/living/silicon/robot/drone/drone.dm [hunk 2]
```diff
@@ -198,8 +194,7 @@
 	emagged = 1
 	density = 1
 	pass_flags = 0
-	icon_state = "repairbot-emagged"
-	holder_type = /obj/item/holder/drone/emagged
+	icon_state = "[icon_state]-emag" //applies emag-specific sprites based on current chassis
```
Comment (Kyep) [inline code/modules/mob/living/silicon/robot/drone/drone.dm:197]: Emagging a drone twice won't result in an invalid icon_state?

File: code/modules/mob/living/silicon/robot/drone/drone.dm [hunk 3]
```diff
@@ -297,18 +292,16 @@
...
 	var/icontype
-	icontype = input(player,"Pick an icon") in sprite
-	icon_state = sprite[icontype]
-	updateicon()
-
-	choose_icon(6,sprite)
-*/
-
+	icontype = input(player,"Choose a chassis design.") in drone_icons
+	icon_state = drone_icons[icontype]
```
Comment (Kyep) [inline code/modules/mob/living/silicon/robot/drone/drone.dm:303]: What if a player selects 'cancel' or closes the window?

File: code/modules/mob/living/silicon/robot/drone/drone_abilities.dm [hunk 4]
```diff
@@ -49,9 +49,9 @@
 	else
 		..()
 
-/mob/living/silicon/robot/drone/verb/customize()
+/mob/living/silicon/robot/drone/verb/customize()	//snowflake time! if our drone player has a drone specific fluff sprite, they can use this verb to activate it.
```
Comment (Kyep) [inline code/modules/mob/living/silicon/robot/drone/drone_abilities.dm:52]:
```suggestion
/mob/living/silicon/robot/drone/verb/customize()	// Players with their own fluff drone sprites, can activate them using this verb.
```

File: code/modules/mob/living/silicon/robot/drone/drone.dm [hunk 5]
```diff
@@ -297,18 +292,16 @@
 	to_chat(src, "<b>Don't invade their worksites, don't steal their resources, don't tell them about the changeling in the toilets.</b>")
 	to_chat(src, "<b>Make sure crew members do not notice you.</b>.")
 
-/*
-	sprite["Default"] = "repairbot"
-	sprite["Mk2 Mousedrone"] = "mk2"
-	sprite["Mk3 Monkeydrone"] = "mk3"
+	//now that the drone has a player, let's pick a sprite.
+	var/drone_icons[0]	//these are our sprite choices
+	drone_icons["Standard"] = "repairbot"
```
Comment (Kyep) [inline code/modules/mob/living/silicon/robot/drone/drone.dm:297]: Its weird that all the sprites are named drone_something, except the default one.

File: code/modules/mob/living/silicon/robot/drone/drone.dm [hunk 6]
```diff
@@ -100,17 +100,13 @@
 /mob/living/silicon/robot/drone/get_default_name()
 	return "maintenance drone ([rand(100,999)])"
 
-/mob/living/silicon/robot/drone/update_icons()
+/mob/living/silicon/robot/drone/update_icons() //sets our eyes correctly based on emag status and chassis
```
Comment (Kyep) [inline code/modules/mob/living/silicon/robot/drone/drone.dm:103]:
```suggestion
/// Set eye icons, based on consciousness, chassis and emag status.
/mob/living/silicon/robot/drone/update_icons() 
```

**Comment by taukausanake** [comment]
> * The green color is dubious too. We already have diona nypmhs, small green things that scurry around

That's a false statement. Diona are mostly brown with a green eye. No one is going to get those mixed up.

@ExusA You said these are a port. Where are they originally from? Would be good to give credit where due and make sure it's okay to port from elsewhere

File: code/modules/mob/living/silicon/robot/drone/drone.dm (hunk 2 above)
Comment (ExusA) [reply code/modules/mob/living/silicon/robot/drone/drone.dm:197]: emag_act() returns before we get to the icon state reassignment if the drone is already emagged, unless i'm missing something

File: code/modules/mob/living/silicon/robot/drone/drone.dm (hunk 3 above)
Comment (ExusA) [reply code/modules/mob/living/silicon/robot/drone/drone.dm:303]: icon_state is set to 'repairbot' inherently, so cancelling the selection will just leave them with the standard drone sprite

File: code/modules/mob/living/silicon/robot/drone/drone.dm (hunk 5 above)
Comment (ExusA) [reply code/modules/mob/living/silicon/robot/drone/drone.dm:297]: i was thinking that too, can change it now

**Comment by ExusA** [comment]
@Kyep 
- Since we don't have the clockwork cult in our codebase, I think a clockwork styled drone is more of a nod to it than confusing. The average player isn't going to see a clockwork drone and think there's a cult aboard. In any case, I'll change the name to be less associated with the cult.
- Purple also means janitorial, which is a big part of what drones do.
- Like has been said, diona nymphs are quite different colour and shape wise, nobody's going to get confused with them.
- Yeah, they are :p not sure how you'd make them look much clearer with so few pixels to work with, any suggestions there?

@taukausanake the green pointy one is originally from TG,  and the other two are from Skyrat.

File: code/modules/mob/living/silicon/robot/drone/drone.dm (hunk 5 above)
Comment (ExusA) [reply code/modules/mob/living/silicon/robot/drone/drone.dm]: changed all references to drone_repairbot

File: code/modules/mob/living/silicon/robot/drone/drone.dm (hunk 1 above)
Comment (ExusA) [reply code/modules/mob/living/silicon/robot/drone/drone.dm]: done!

**Comment by Pixelzilla** [comment]
Not a fan of having customization at all for Maintenance Drones. Them being flashy and noticeable instead of fairly bland and uniform looking kinda defeats their entire point and the rules surrounding playing them. Its a little bit harder to follow the rules of playing one like "Make sure crew members do not notice you.." if you have customization options. Customization options are naturally made so that someone stands out and is noticed more.

**Comment by Dumbdumn5** [comment]
Having a minor customization option for a role that's just to dick around with building and repairing things as a ghost is just fine to me! It's not a big change and it's pretty enjoyable to have something a little bit fresh. The eye colors being consistent are solid and while I can understand the concern with the head sprites, having made a few before then it's not exactly practical to have the things look one for one the same as the original drones. 

As far as confusion goes, have you ever personally seen anything that looks remotely like these drone sprites anywhere else in the game? Low to the ground, neon blue or red eyes, scuttling about, etc.

In no way shape or form am I gonna look at that runt and go 'damn that's a tiny wizard' or consider the thing a nymph, and even if I did consider it the latter, why would that change just about anything as far as the way I interact with the mostly inconsequential ghost role that requires adherence to OOCly enforced lawsets? @Kyep

**Comment by AffectedArc07** [comment]
Im not super sold on the idea of giving maintenance drones extra customisation. Drones should be easy to spot for the crew, emagged or not. These two also look like semi-hostile creates in my opinion
image

Thats just my opinion though.

**Comment by ExusA** [comment]
I think ultimately it has the _potential_ to confuse _some_ players, as with pretty much any sprite that isn't obviously something you're familiar with in another context. There are an overwhelming amount of objects in SS13 that aren't self-explanatory just from their sprites. Some examples I can think of are, Holobarrier projectors, half the food sprites, half the cyborg chassis, the cyborg charger, the list goes on. For, heck, half the objects in the game, you have to examine them or at least mouseover to get a clearer idea or even any indication at all of what they might be. I think after the first time a player sees each sprite, they'll just know it's a drone, in the same way you know what an RPD is and does despite nothing about its appearance revealing the fact. If someone sees something they haven't seen before, they usually examine it. Doing so makes it as clear as possible that it's not a hostile creature.

**Comment by ExusA** [comment]
As much as it pains me, I need to close this for a bit because _something_ went wrong _somewhere_ and i destroyed my fork and everything on it in a way I don't know how to fix, so i'll just remake and reopen this when I get the time, hopefully with better sprites to boot.

**Pull request closed without merging**